internal/service/arcregionswitch/plan.go

```diff
@@ -0,0 +1,1373 @@
+
+	// Handle tags update
+	if !plan.TagsAll.Equal(state.TagsAll) {
+		if err := UpdateTags(ctx, conn, plan.ID.ValueString(), state.TagsAll, plan.TagsAll); err != nil {
+			resp.Diagnostics.AddError("updating tags", err.Error())
+			return
+		}
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (r *resourcePlan) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state resourcePlanModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.Meta().ARCRegionSwitchClient(ctx)
+
+	input := arcregionswitch.DeletePlanInput{
+		Arn: aws.String(state.ID.ValueString()),
+	}
+
+	_, err := conn.DeletePlan(ctx, &input)
+	if err != nil {
+		resp.Diagnostics.AddError("deleting ARC Region Switch Plan", err.Error())
+		return
+	}
+}
+
+func (r *resourcePlan) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func (r *resourcePlan) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	// Basic validation is handled by the schema validators
+}
+
+func (r *resourcePlan) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = request.ProviderTypeName + "_arcregionswitch_plan"
+}
+
+func (r *resourcePlan) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = fwschema.Schema{
+		Attributes: map[string]fwschema.Attribute{
+			names.AttrARN: framework.ARNAttributeComputedOnly(),
+			names.AttrID:  framework.IDAttribute(),
+			names.AttrName: fwschema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"execution_role": fwschema.StringAttribute{
+				Required: true,
+				Validators: []validator.String{
+					fwvalidators.ARN(),
+				},
+			},
+			"recovery_approach": fwschema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf("activeActive", "activePassive"),
+				},
+			},
+			"regions": fwschema.ListAttribute{
+				Required:    true,
+				ElementType: types.StringType,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
+				},
+			},
+			names.AttrDescription: fwschema.StringAttribute{
+				Optional: true,
+			},
+			"primary_region": fwschema.StringAttribute{
+				Optional: true,
+			},
+			"recovery_time_objective_minutes": fwschema.Int64Attribute{
+				Optional: true,
+			},
+			names.AttrTags:    tftags.TagsAttribute(),
+			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+		},
+		Blocks: map[string]fwschema.Block{
+			"associated_alarms": fwschema.SetNestedBlock{
```

Comment:
Adding the appropriate `CustomType` declarations to the schema and corresponding data structure will allow AutoFlex to handle expanding/flattening in place of custom functions.

```suggestion
			"associated_alarms": fwschema.SetNestedBlock{
			         CustomType: fwtypes.NewSetNestedObjectTypeOf[associatedAlarmModel](ctx),

```